### PR TITLE
Chapter1/Homework/8

### PR DESCRIPTION
**Pull request by b4nter**

### CODE REVIEW

File: Src/BootCamp.Chapter/FileCleaner.cs [hunk 1]
```diff
@@ -13,8 +14,48 @@ public static class FileCleaner
...
+
+        public static void ValidateName(string fileContent)
         {
-            File.WriteAllText(cleanedFile, "a");
+            var name = BalanceStats.ConvertToArray(fileContent)[0];
+            foreach (char letter in name)
+            {
+                if (!Char.IsLetter(letter) && !letter.Equals(' '))
+                {
+                    throw new Exception();
```
Comment (Almantask) [inline Src/BootCamp.Chapter/FileCleaner.cs]: Codacy Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
- Added 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/FileCleaner.cs  3
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948&bid=16532595)

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 2]
```diff
@@ -7,31 +10,271 @@ public static class BalanceStats
...
+        public static float GetRichestPersonsBalance(string[] peopleAndBalances)
+        {
+            if (peopleAndBalances.Length == 1)
+            {
+                return GetCurrentBalancefor(peopleAndBalances[0]);
+            }
+            var richestPersonsBalance = GetCurrentBalancefor(peopleAndBalances[0]);
+            for (int i = 0; i < peopleAndBalances.Length - 1; i++)
+            {
+          //      richestPersonsBalance = GetCurrentBalancefor(peopleAndBalances[i]);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948)

File: Src/BootCamp.Chapter/Program.cs [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
+
+            Part2();
+        }
+
+        static void Part2()
         {
+            var fileContent = File.ReadAllText(@"Input\Balances.clean");
+            var peopleAndBalances = fileContent.Split(Environment.NewLine);
+
+            //    BalanceStats.ConvertToBalanceArray(peopleAndBalances[0]);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/FileCleaner.cs  3
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948&bid=16532595)

**Comment by b4nter** [comment]
Locally it passes all tests, not sure what went wrong here.

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 4]
```diff
@@ -7,31 +10,270 @@ public static class BalanceStats
...
+                }
+            }
+            // To remove comma at the end.
+            var peopleWithSameBalance = sb.ToString().Remove(sb.ToString().Length - 2);
+            return ConvertToArray(peopleWithSameBalance);
+        }
+
+        public static float GetHighestBalanceEver(string[] peopleAndBalances)
+        {
+            var highestBalance = GetHighestBalanceFor(peopleAndBalances[0]);
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: It's great that you managed to reuse some of the already made code.

File: Src/BootCamp.Chapter/BalanceStats.cs [hunk 5]
```diff
@@ -7,31 +10,270 @@ public static class BalanceStats
...
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                if (highestBalance < GetHighestBalanceFor(peopleAndBalances[i]))
+                {
+                    highestBalance = GetHighestBalanceFor(peopleAndBalances[i]);
+                }
+            }
+            return highestBalance;
+        }
+        public static float GetHighestBalanceFor(string personAndBalance)
```
Comment (Almantask) [inline Src/BootCamp.Chapter/BalanceStats.cs]: "For" suffix is probably not needed. Though in some cases such style works. It's a matter of taste, I personally wouldn't write it.

File: Src/BootCamp.Chapter/Program.cs [hunk 6]
```diff
@@ -1,16 +1,27 @@
...
     {
         static void Main(string[] args)
+        {   
+            //part 1
+            FileCleaner.Clean(@"Input\Balances.corrupted", @"Input\Balances.clean");
+
+            Part2();
+        }
+
+        static void Part2()
```
[the remark this answers is not in the document]
Comment (b4nter) [reply Src/BootCamp.Chapter/Program.cs]: In this case I would disagree that it needs renaming, in homework 6 there was method called very similarly. Also in practice there would be no real use for such a method it is just to showcase part2 functionality.

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/FileCleaner.cs  3
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948&bid=16532595)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/FileCleaner.cs  3
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948&bid=16532595)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/FileCleaner.cs  3
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948&bid=16532595)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 3
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/FileCleaner.cs  3
- Src/BootCamp.Chapter/BalanceStats.cs  4
- Src/BootCamp.Chapter/TextTable.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948&bid=16532595)

**Pull request closed without merging**